pkg/status/combinedstatus-resolution.go

```diff
@@ -357,32 +377,34 @@ func (c *combinedStatusResolution) evaluateWorkStatus(celEvaluator *celEvaluator
 // The function assumes that the caller holds a lock over the combinedstatus
 // resolution.
 func evaluateWorkStatusAgainstStatusCollectorWriteLocked(celEvaluator *celEvaluator, workStatusWECName string,
-	workStatusContent map[string]interface{}, scData *statusCollectorData) (bool, error) {
+	workStatusContent map[string]interface{}, scData *statusCollectorData) bool {
 	wsData, exists := scData.wecToData[workStatusWECName]
 
 	if workStatusContent == nil && exists { // workstatus is empty/deleted, remove the workstatus data if it exists
 		delete(scData.wecToData, workStatusWECName)
-		return true, nil
+		return true
 	}
 
 	// evaluate filter to determine if the workstatus is relevant
 	if scData.Filter != nil {
 		eval, err := celEvaluator.Evaluate(*scData.Filter, workStatusContent)
 		if err != nil {
-			return false, fmt.Errorf("failed to evaluate filter expression: %w", err)
+			runtime2.HandleError(fmt.Errorf("failed to evaluate filter expression: %w", err))
```

Comment:
FYI, in Kubernetes release 1.31 we get https://github.com/kubernetes/apimachinery/blob/v0.31.0-alpha.1/pkg/util/runtime/runtime.go#L159-L180

Reply:
This will be addressed in PR for error handling.

Reply:
I do not think we want to hold up that PR until after IBM Cloud offers a version of OpenShift that uses Kubernetes release 1.31 (which has not come out yet).